libs/vkd3d/device.c

```diff
@@ -5106,17 +5121,109 @@ static HRESULT STDMETHODCALLTYPE d3d12_device_OpenSharedHandleByName(d3d12_devic
 static HRESULT STDMETHODCALLTYPE d3d12_device_MakeResident(d3d12_device_iface *iface,
         UINT object_count, ID3D12Pageable * const *objects)
 {
-    FIXME_ONCE("iface %p, object_count %u, objects %p stub!\n",
-            iface, object_count, objects);
+    struct d3d12_device *device = impl_from_ID3D12Device(iface);
+    const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+
+    if (device->vk_info.EXT_pageable_device_local_memory)
+    {
+        uint32_t i;
+        
+        for (i = 0; i < object_count; i++)
+        {
+            ID3D12Heap *heap_iface;
+            ID3D12Resource *resource_iface;
+            ID3D12DeviceChild *object = (ID3D12DeviceChild*)objects[i];
```

Comment:
Why not just QueryInterface from the ID3D12Pageable?

Reply:
Dunno, will look into it :grinning: